src/main/java/com/dmdirc/ui/messages/EventFormatter.java

```diff
@@ -155,4 +155,15 @@ private String applyDisplayProperties(final DisplayPropertyMap displayProperties
         return res.toString();
     }
 
+    /**
+     * Get the displayable properties map for this event according to the formatter.
+     *
+     * @param event Event to get properties for.
+     * @return DisplayPropertyMap from formatter.
+     */
+    public DisplayPropertyMap getFormatDisplayableProperties(final DisplayableEvent event) {
```

Comment:
I don't really like this coming from EventFormatter. It's job is to format, not provide information about random events.

Can you just get it from the provider directly?